bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/handler/BroadlinkBaseThingHandler.java

```diff
@@ -0,0 +1,293 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.broadlink.handler;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.broadlink.BroadlinkBindingConstants;
+import org.openhab.binding.broadlink.config.BroadlinkDeviceConfiguration;
+import org.openhab.binding.broadlink.internal.*;
+import org.openhab.binding.broadlink.internal.discovery.DeviceRediscoveryAgent;
+import org.openhab.binding.broadlink.internal.discovery.DeviceRediscoveryListener;
+import org.openhab.binding.broadlink.internal.socket.NetworkTrafficObserver;
+import org.openhab.binding.broadlink.internal.socket.RetryableSocket;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.util.HexUtils;
+import org.slf4j.Logger;
+
+/**
+ * Abstract superclass of all supported Broadlink devices.
+ *
+ * @author John Marshall/Cato Sognen - Initial contribution
+ */
+@NonNullByDefault
+public abstract class BroadlinkBaseThingHandler extends BaseThingHandler implements DeviceRediscoveryListener {
+
+    private static final String INITIAL_DEVICE_ID = "00000000";
+
+    @Nullable
+    private RetryableSocket socket;
+
+    @Nullable // Can be injected for test purposes
+    private NetworkTrafficObserver networkTrafficObserver;
+
+    private int count;
+
+    protected BroadlinkDeviceConfiguration thingConfig;
+    protected final ThingLogger thingLogger;
```

Comment:
```suggestion
public final Logger logger = LoggerFactory.getLogger(getClass());
```
Any reason why you use this for logging instead of org.slf4j.Logger? This one line is all that is needed, you dont need to initilize it in the constructor.

Also group all the final variables together at the top before any non final.